### PR TITLE
updated dependencies and gradles

### PR DESCRIPTION
**Pull request by TacoTheDank**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -46,22 +46,22 @@ android {
...
     implementation group: 'org.javatuples', name: 'javatuples', version: '1.2'
-    implementation 'com.fasterxml.jackson.core:jackson-databind:2.8.11.1'
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.9.6'
 
     implementation 'ch.acra:acra:4.11'
 
-    testCompile 'junit:junit:4.12'
-    testCompile "org.robolectric:robolectric:3.6.1"
+    testImplementation 'junit:junit:4.12'
+    testImplementation "org.robolectric:robolectric:4.0-alpha-2"
```
Comment (brarcher) [inline app/build.gradle]: I would like to avoid using alpha releases. Should we wait on this one?

Comment (TacoTheDank) [reply app/build.gradle]: ill set it to 3.8, which is the latest stable

**Comment by TacoTheDank** [comment]
ill try 23.0 for guava, since its the latest version that doesn't include a "-[word]" after it

**Comment by brarcher** [comment]
Curious, what is the drive for updating the dependencies? There are no CVEs out against these that would affect the application. Do you have subsequent changes planned which need these?

**Comment by brarcher** [comment]
Have you had a chance to test it locally for regressions? The automated tests check that an ENCODE intent can be received and a file can be encoded, but the general UI is not tested at this point.

**Comment by TacoTheDank** [comment]
oh. no i havent

**Comment by brarcher** [comment]
Are you in a position where you can do a spot check, to see if with the updates you are able to select a video file, play with the sliders, and encode a file?

**Comment by TacoTheDank** [comment]
i mean, ill try...
ill build an apk and test it on my phone

**Comment by TacoTheDank** [comment]
i built a debug and a release apk from my fork - ill put the build file here in case you'd like to test it as well
(fuck you github and your 10MB limit >:c i have to put it on a cloud and make it shareable now)

https://mega.nz/#F!0TA2zAQB!U2P08W5sdOlXuBMGfUWbOw

**Comment by brarcher** [comment]
lol, yeah, I know the apk is pretty big.

Just let me know if you find any issues. If there are, we'll need to figure them out. If not, I'll accept the PR.

**Comment by TacoTheDank** [comment]
kk

**Comment by TacoTheDank** [comment]
(also i uploaded the apks above if you didnt see that, cause i was editing that comment when you commented)

**Comment by brarcher** [comment]
Did you find anything in your testing?

**Comment by brarcher** [comment]
I tried it out as well. I found that Android Studio did not like to build it due to an incompatibility with the Gradle version and the Android plugin. Getting around that, the app itself seemed to work just fine.

I pushed my workaround. Once that passes, I'll merge the request.

**Comment by TacoTheDank** [comment]
oh yeah the configure on demand thing

**Pull request merged**